.github/workflows/tests.yaml

```diff
@@ -0,0 +1,69 @@
+name: Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  linters:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install python packages and dependencies
+        run: |
+          pip install -U .[dev]
+      - name: Run black
+        run: |
+          make test-black
+      - name: Run codespell
+        run: |
+          make test-codespell
+      - name: Run flake8
+        run: |
+          make test-flake8
+      - name: Run isort
+        run: |
+          make test-isort
+      - name: Run mypy
+        run: |
+          make test-mypy
+      - name: Run pydocstyle
+        run: |
+          make test-pydocstyle
+      - name: Run pyright
```

Comment:
Pylint seems to be missing, is it left out intentionally?

Reply:
added